### ISSUE TITLE
Not working with Sensors from Protheus

### ISSUE THREAD

**Issue by Evolver13** [alexiosc/ecofrog#2]
Devices that are not bought from e-sensorix will not work as the API-Adress is not the same. 

Can you please take a look into the Protheus API also?

**Comment by Evolver13** [comment]
[2023-03-13_Azure_API Beschreibung_v02.pdf](https://github.com/alexiosc/ecofrog/files/10961365/2023-03-13_Azure_API.Beschreibung_v02.pdf)

This is the API-Instruction for Protheus. 

i Tried and change some variables in the files but cant get it working.

**Comment by alexiosc** [comment]
Would you happen to know the correct API address for Protheus devices?

**Comment by Evolver13** [comment]
Hi, 

I think it's not only to he API.

I got the sensors working with the following code.

I Discribed it in the HA-Forum.
If you need a tester for your integration with the new API send me a DM.

https://community.home-assistant.io/t/proteus-sensor-api-integration/463144/12?u=evolver13
sensor:
  - pl
      

sensor:
  - platform: rest
    scan_interval: 86400
    name: ecofrog
    json_attributes:
      - DeviceID
      - DeviceName
      - Medium
      - LastUpdate
      - MaxVolume
      - ActualVolume
      - ActualVolumePercent
      - Battery
      - Temperature
      - MessData
    resource: https://devapiext.azurewebsites.net/api/latest?code=qRX_UuM9UYPUuPV5B4lqEjUuzcpf_UrzDi40ReHSTTL7AzFu_4PDnA==
    payload: '{DeviceRequest: {"Username": "xy@proteus.news", "accountID": "12345", "SecurityStamp": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "DeviceID": "xxxxxxx" }}'
    value_template: "OK"
  - platform: template
    sensors:
      eco_id:
        friendly_name: "Eco_DeviceID"
        value_template: "{{ state_attr('sensor.ecofrog', 'DeviceID') }}"
      eco_name:
        friendly_name: "Eco_Name"
        value_template: "{{ state_attr('sensor.ecofrog', 'DeviceName') }}"
      eco_medium:
        friendly_name: "Eco_Medium"
        value_template: "{{ state_attr('sensor.ecofrog', 'Medium') }}"
      eco_lastupdate:
        friendly_name: "Eco_LastUpdate"
        value_template: "{{ state_attr('sensor.ecofrog', 'LastUpdate') }}"
      eco_maxmolume:
        friendly_name: "Eco_MaxVolume"
        value_template: "{{ state_attr('sensor.ecofrog', 'MaxVolume') }}"
        unit_of_measurement: "L"
      eco_actualvolume:
        friendly_name: "Eco_ActualVolume"
        value_template: "{{ state_attr('sensor.ecofrog', 'ActualVolume') }}"
        unit_of_measurement: "L"
      eco_actualvolumepercent:
        friendly_name: "Eco_ActualVolumePercent"
        value_template: "{{ state_attr('sensor.ecofrog', 'ActualVolumePercent') }}"
        unit_of_measurement: "%"
      eco_battery:
        friendly_name: "Eco_Battery"
        value_template: "{{ state_attr('sensor.ecofrog', 'Battery') }}"
        unit_of_measurement: "V"
      eco_temperature:
        friendly_name: "Eco_Temperature"
        value_template: "{{ state_attr('sensor.ecofrog', 'Temperature') }}"
        unit_of_measurement: "°C"
      eco_messdata:
        friendly_name: "Eco_MessData"
        value_template: "{{ state_attr('sensor.ecofrog', 'MessData') }}"
        unit_of_measurement: "cm"

**Comment by Evolver13** [comment]
Also I made a helper script to fetch the sensor. 
In the sensor code above the scan Intervall must be raised to a year or so to prevent to many API requests

https://community.home-assistant.io/t/proteus-sensor-api-integration/463144/14?u=evolver13

**Comment by alexiosc** [comment]
@Evolver13 you're right, it's not just the API. The Proteus API endpoint returns proper JSON! In fact, I only made my integration because my e-sensorix service returns this data enclosed in a string, so the JSON config needs to be parsed twice: once to get a string, and another one to decode the structure represented **in** the string. If this weren't the case, I would have made precisely what you did with the REST integration.

**Comment by Evolver13** [comment]
@alexiosc as i mentioned in my HA Community post maybe its possible to integrate my approach into your add-on. iam far away from being able to alter your integration to fit both types of sensors. .
As for now its working fine for me but its not quite as nice or easy to setup like yours.

**Comment by razem-io** [comment]
Hi, I created an addon for EcoFrog devices. Currently, only supporting proteus-sensor.de devices, which this issue is about, I think. It should be easy to setup, since you do not need the API credentials from the provider. Just the credentials you use to access the cloud dashboard.

https://github.com/razem-io/ha-ecofrog-cloud
